src/core/ai/llm/messages/manager.ts

```diff
@@ -525,7 +525,17 @@ export class ContextManager {
         try {
             // Use pre-computed system prompt if provided
             const prompt = systemPrompt ?? (await this.getSystemPrompt(context));
-            return this.formatter.format([...messageHistory], prompt, context);
+
+            // Map DynamicContributorContext to FormatterContext
+            const formatterContext: FormatterContext | undefined = context.llmProvider
```

Comment:
nit: we should rechange dynamiccontributor to use provider/model convention. marking for my reference